stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,25 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+      self = ""
+      return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
-    for _ in 0..<count {
-      result += repeatedValue
+    var repeatedValue = repeatedValue
+    self = repeatedValue.withUTF8 { repeatedUTF8 in
+      String(_uninitializedCapacity: repeatedUTF8.count * count) { buffer in
+        var total = 0
+        for i in 0..<count {
+          let offset = i * repeatedUTF8.count
```

Comment:
Does this make sense? Is wrapping arithmetic really faster on modern CPUs?

Reply:
It's not so much that the wrapping arithmetic is inherently faster as it is that the error path on the non-wrapping arithmetic inhibits a lot of optimizations.

Reply:
Could `&*` be used on line 43 as well as on 46?

Reply:
Capacity of any collection is limited to `Int`. Why then there's no rule to use `&*` and `&+` everywhere in stdlib if it's so beneficial? I am not trolling. I see in some places `&*` is used. Not everywhere but in some places. Why exactly in such places? I have no idea. May be any documentation exists on this matter?

Reply:
You _must_ check for overflow on line 43: it can overflow because the user can ask for too many repeats. However, if line 43 doesn't overflow, then line 46 can't overflow. That's why you can eliminate the overflow check.

Reply:
Oh! Now I get the idea of where to use saturated arithmetic!

Reply:
Fixed. Thank you for suggestion.